snakemake_storage_plugin_s3/__init__.py

```diff
@@ -51,11 +51,19 @@ class StorageProviderSettings(StorageProviderSettingsBase):
             # For other items, we rather recommend to let people use a profile
             # for setting defaults
             # (https://snakemake.readthedocs.io/en/stable/executing/cli.html#profiles).
-            "env_var": False,
+            "env_var": True,
             # Optionally specify that setting is required when the executor is in use.
             "required": False,
         },
     )
+    aws_region: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "AWS S3 region constraint for AWS S3 storage requirement",
+            "env_var": True,
+            "required": False,
+        },
+    )
```

Comment:
Below, the region is passed as a generic LocationConstraint. Hence, I would suggest to name it similarly here, as this plugin can also deal with S3 other than the amazon flavor.

```suggestion
    location: Optional[str] = field(
        default=None,
        metadata={
            "help": "Location constraint for the storage (e.g. AWS S3 region)",
            "env_var": True,
            "required": False,
        },
    )
```